### PR TITLE
Fix minor errors in 4.x

### PR DESCRIPTION
**Pull request by deepakrkris**
Fix coding errors in 4.x branch
- [ ] - fix spelling for 'workspaceId'
- [ ] - load all built-in models
- [ ] - pass facetName as argument for model-config

### CODE REVIEW

**Comment by superkhau** [comment]
You should add the above bullet list *inside* the commit message body.

File: common/models/model-config.js [hunk 1]
```diff
@@ -3,6 +3,7 @@
 // This file is licensed under the MIT License.
 // License text available at https://opensource.org/licenses/MIT
 'use strict';
+const clone = require('lodash').clone;
```
Comment (superkhau) [inline common/models/model-config.js]: Object.assign

File: common/models/model-config.js [hunk 2]
```diff
@@ -17,10 +18,18 @@ module.exports = function(ModelConfig) {
         cb = options;
         options = {};
       }
-      const id = data.id;
+      const modelConfig = clone(data);
```
Comment (superkhau) [inline common/models/model-config.js]: ditto

File: component/workspace.js [hunk 3]
```diff
@@ -24,9 +24,12 @@ class Workspace extends Graph {
     mixin(this, Tasks.prototype);
     this.middlewarePhases = [];
   }
-  addBuiltInModel(name) {
-    // add ModelDefinition to the workspace graph for a builtin loopback model
-    new Model(this, name, {}, {builtIn: true});
+  addBuiltInModels(list) {
+    const workspace = this;
```
Comment (superkhau) [inline component/workspace.js]: pass in this directly

File: component/workspace.js [hunk 4]
```diff
@@ -24,9 +24,12 @@ class Workspace extends Graph {
...
+    list.forEach(function(name) {
+      // add ModelDefinition to the workspace graph for a builtin loopback model
+      new Model(workspace, name, {}, {builtIn: true});
```
Comment (superkhau) [inline component/workspace.js]: pass in this directly

File: connector/template-handler.js [hunk 5]
```diff
@@ -1,32 +1,39 @@
 'use strict';
 
 const fs = require('fs-extra');
+const clone = require('lodash').clone;
```
Comment (superkhau) [inline connector/template-handler.js]: Object.assign

File: connector/template-handler.js [hunk 6]
```diff
@@ -37,13 +44,18 @@ class TemplateHandler {
...
+              modelConfig,
+              next);
           });
         });
       }
       if (facet.middleware) {
-        facet.middleware.forEach(function(configData) {
+        facet.middleware.forEach(function(middleware) {
           taskList.push(function(next) {
+            let configData = clone(middleware);
```
Comment (superkhau) [inline connector/template-handler.js]: ditto

File: test/acceptance-tests/use-cases/test-scripts/create-model-config.js [hunk 7]
```diff
@@ -69,7 +71,6 @@ module.exports = function() {
   this.Then(/^the model config is returned$/, function(next) {
     expect(Object.keys(testsuite.modelConfig)).to.include.members([
```
Comment (superkhau) [inline test/acceptance-tests/use-cases/test-scripts/create-model-config.js]: check for the single key instead

`expect(testsuite.modelConfig).to.have.key('dataSource');`

File: test/acceptance-tests/use-cases/test-scripts/create-model-config.js [hunk 8]
```diff
@@ -48,6 +48,8 @@ module.exports = function() {
       if (err) return next(err);
       const config = data[testsuite.modelId];
       expect(config).to.not.to.be.undefined();
+      delete testsuite.ModelConfig.id;
+      delete testsuite.ModelConfig.facetName;
```
Comment (superkhau) [inline test/acceptance-tests/use-cases/test-scripts/create-model-config.js]: We should remove the assertion on line 24, you should only be asserting in `this.Then`. The lines here probably need to be moved up into `this.Given` as this is part of your test setup.

Comment (deepakrkris) [inline test/acceptance-tests/use-cases/test-scripts/create-model-config.js]: Given is also a condition

File: connector/template-handler.js (hunk 6 above)
Comment (deepakrkris) [inline connector/template-handler.js]: agreed

File: connector/template-handler.js (hunk 5 above)
Comment (deepakrkris) [inline connector/template-handler.js]: agreed

File: common/models/model-config.js (hunk 1 above)
Comment (deepakrkris) [inline common/models/model-config.js]: agreed

File: common/models/model-config.js (hunk 2 above)
Comment (deepakrkris) [inline common/models/model-config.js]: agreed

File: component/workspace.js (hunk 3 above)
Comment (deepakrkris) [inline component/workspace.js]: agreed

File: component/workspace.js (hunk 4 above)
Comment (deepakrkris) [inline component/workspace.js]: agreed

File: test/acceptance-tests/use-cases/test-scripts/create-model-config.js (hunk 7 above)
Comment (deepakrkris) [inline test/acceptance-tests/use-cases/test-scripts/create-model-config.js]: agreed

File: test/acceptance-tests/use-cases/test-scripts/create-model-config.js [hunk 9]
```diff
@@ -44,11 +44,14 @@ module.exports = function() {
...
+    const expectedConfig = Object.assign(testsuite.ModelConfig);
+    delete expectedConfig.id;
+    delete expectedConfig.facetName;
     fs.readJson(file, function(err, data) {
       if (err) return next(err);
-      const config = data[testsuite.modelId];
-      expect(config).to.not.to.be.undefined();
-      expect(testsuite.ModelConfig).to.eql(config);
+      const storedConfig = data[testsuite.modelId];
+      expect(storedConfig).to.not.to.be.undefined();
```
Comment (superkhau) [inline test/acceptance-tests/use-cases/test-scripts/create-model-config.js]: delete

**Pull request merged**